src/components/Modal/BaseModal.js

```diff
@@ -76,20 +78,33 @@ function BaseModal({
     );
 
     useEffect(() => {
-        Modal.willAlertModalBecomeVisible(isVisible);
-
+        if (!isVisible) {
+            return;
+        }
+        Modal.willAlertModalBecomeVisible(true);
         // To handle closing any modal already visible when this modal is mounted, i.e. PopoverReportActionContextMenu
-        Modal.setCloseModal(isVisible ? onClose : null);
+        Modal.setCloseModal(onClose);
     }, [isVisible, onClose]);
 
+    useEffect(() => {
+        isVisibleRef.current = isVisible;
+        return () => {
+            if (!isVisible) {
+                return;
+            }
+            Modal.willAlertModalBecomeVisible(false);
+            Modal.setCloseModal(null);
```

Comment:
I don't think this logic fits best as the cleanup logic. Can we done this using usePrevious instead?

```js
    useEffect(() => {
        if (isVisible) {
            Modal.willAlertModalBecomeVisible(true);
            Modal.setCloseModal(onClose);
        } else if (wasVisible && !isVisible) {
            Modal.willAlertModalBecomeVisible(false);
            Modal.setCloseModal(null);
        }
    }, [isVisible, wasVisible, onClose]);

```